script/install.sh

```diff
@@ -354,16 +355,19 @@ run_command() {
                 install_compiler "$2"
             fi
 
-            local -r binpath=$(binpath_for_compiler "$2")
-            if [ -f "$ROOT/$2/$binpath/dub" ]; then
-                if [[ $("$ROOT/$2/$binpath/dub" --version) =~ ([0-9]+\.[0-9]+\.[0-9]+(-[^, ]+)?) ]]; then
-                    log "Using dub ${BASH_REMATCH[1]} shipped with $2"
+
+            if ! [[ "$2" =~ ^dub ]]; then
+                local -r binpath=$(binpath_for_compiler "$2")
+                if [ -f "$ROOT/$2/$binpath/dub" ]; then
+                    if [[ $("$ROOT/$2/$binpath/dub" --version) =~ ([0-9]+\.[0-9]+\.[0-9]+(-[^, ]+)?) ]]; then
+                        log "Using dub ${BASH_REMATCH[1]} shipped with $2"
+                    else
+                        log "Using dub shipped with $2"
+                    fi
                 else
-                    log "Using dub shipped with $2"
+                    DUB_BIN_PATH="${ROOT}/dub"
+                    install_dub
                 fi
```

Comment:
I deliberately avoided updating the default `dub` symlink (which happens in `install_dub`), though there are pros and cons for this.